### ISSUE TITLE
Support the "official" debug gem?

### ISSUE THREAD

**Issue by iandol** [rubyide/vscode-ruby#780]
### Your environment

- `vscode-ruby` version: V0.28.1
- Ruby version: 3.0.2
- Ruby version manager (if any): rbenv
- VS Code version: 1.62.3
- Operating System: macOS 12.0.1
- Using language server? true

### Request

There is a newer `debug` gem <https://github.com/ruby/debug> that offers a very comprehensive feature set, and a new VScode extension demonstrates it in action without any other extensions required: https://marketplace.visualstudio.com/items?itemName=KoichiSasada.vscode-rdbg — specifically being able to remote debug code triggered from within-the-code-itself works great. It would be nice to explore switching the debugger over to the official extension as it has more features and appears to integrate well with with VSCode.

**Comment by iandol** [comment]
Note that rails is switching over to `ruby/debug`: https://www.bigbinary.com/blog/rails-7-replaced-byebug-with-ruby-debug

**Comment by iandol** [comment]
Ruby 3.1 released and `ruby/debug` is now officially released: https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/

**Comment by kyrofa** [comment]
Yeah debase doesn't work with ruby 3 at all, right? So ruby-debug-ide is entirely incompatible with ruby 3, isn't it?

**Comment by wingrunr21** [comment]
Tracking branch is `debugger_improvements`
